sdk/ml/azure-ai-ml/azure/ai/ml/entities/_registry/registry_support_classes.py

```diff
@@ -210,10 +178,89 @@ def _to_rest_object(self) -> RestRegistryRegionArmDetails:
                 acr) for acr in self.acr_config]
         storages = []
         if self.storage_config:
-            storages = [SystemCreatedStorageAccount._to_rest_object(
-                storage) for storage in self.storage_config]
+            storages = self._storage_config_to_rest_object()
         return RestRegistryRegionArmDetails(
             acr_details=converted_acr_details,
             location=self.location,
             storage_account_details=storages,
         )
+
+
+    def _storage_config_to_rest_object(self) -> List[RestStorageAccountDetails]:
+        storage = self.storage_config
+        # storage_config can either be a single system-created storage account,
+        # or list of user-inputted id's.
+        if (storage is not None
+            and hasattr(storage, "storage_account_type")
+            and storage.storage_account_type is not None):
+
+            # We DO NOT want to set the arm_resource_id. The backend provides very
+            # unhelpful errors if you provide an empty/null/invalid resource ID,
+            # and ignores the value otherwise. It's better to avoid setting it in
+            # the conversion in this direction at all.
+            # We don't bother processing storage_account_type because the
+            # rest version is case insensitive.
+            account = RestStorageAccountDetails(system_created_storage_account=RestSystemCreatedStorageAccount(
+                storage_account_hns_enabled=storage.storage_account_hns,
+                storage_account_type=storage.storage_account_type,
+            ))
+            # duplicate this value based on the replication_count
+            count = storage.replication_count
+            return [deepcopy(account) for _ in range(0, count)]
+        elif storage is not None and len(storage) > 0:
+            return [make_rest_user_storage_from_id(user_id=user_id) for user_id in storage]
+        else:
+            return []
+
+    @classmethod
```

Comment:
Why does it need to be classmethod?

Reply:
The from rest object methods need to be class methods because there's no entity-registry when this is called. The use case for this function is "We have an Autorest registry class, and we need to convert it into a local entity class registry, so we'll call Registry._from_rest_object(RestRegistry) to do the conversion." That function then recursively calls child class conversion functions, including this.